_posts/2021-04-21-decoding-from-memory-bug.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+published: true
+title:  "Solidity ABI Decoder Bug For Multi-Dimensional Memory Arrays"
+date:   2021-04-21
+author: Solidity Team
+category: Security Alerts
+---
+
+
+On April 5th, 2021, a bug in the Solidity ABI decoder v2 was reported by
+John Toman of the Certora development team. Certora's bug disclosure post
+can be found here: [Memory Isolation Violation in Deserialization Code](https://www.certora.com/blog/deserialization.html).
+
+The bug is fixed with [Solidity version 0.8.4](https://github.com/ethereum/solidity/releases/tag/v0.8.4)
+released on April 21st, 2021. **The bug is present in all prior versions of ABI coder v2.**
+
+We assigned the bug a severity level of "very low", mainly due to the
+fact that it is very hard to exploit the bug.
+
+We are very grateful to John for his continuing efforts to discover bugs in the Solidity compiler.
+
+## Who Should Be Concerned
+
+You could be affected if you are using ABI coder v2 (which is the default starting from 0.8.0) and
```

Comment:
```suggestion
You could be affected if all of the following conditions are met:
 - ABI coder v2 (default starting from 0.8.0) is enabled
 - ``abi.decode`` is used to decode an array that resides in memory
 - array is either multi-dimensional or contains structs
```

Reply:
I think all of them doesn't have to be met. I also think it's rather hard to pinpoint when the bug can happen.

Reply:
I'll merge this for now.